### PR TITLE
 RepositoryDependency id migration step 2: migrate from integer to bigint

### PR DESCRIPTION
**Pull request by tiegz**
* converts `repository_dependencies.id` from integer to bigint 

This migration assumes that we're ok waiting for the table lock, and just temporarily serving bad data for this table until it's done.

### CODE REVIEW

File: db/migrate/20200921173826_convert_repository_dependency_id_to_big_int.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class ConvertRepositoryDependencyIdToBigInt < ActiveRecord::Migration[5.2]
+  def up
+    # NB this locks the table, so we're disabling related code temporarily, while it runs.
+    change_column :repository_dependencies, :id, :bigint
```
Comment (tiegz) [inline db/migrate/20200921173826_convert_repository_dependency_id_to_big_int.rb:4]: @katzj @jsonperl do you know if Circle has a timeout? If so, we could skip this in Circle and run it manually in a long-lived terminal.

Comment (tiegz) [reply db/migrate/20200921173826_convert_repository_dependency_id_to_big_int.rb:4]: Ah nevermind, just remembered we don't deploy via Circle.

File: db/migrate/20200921173826_convert_repository_dependency_id_to_big_int.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  end
+
+  def down
+    change_column :repository_dependencies, :id, :integer
```
Comment (jsonperl) [inline db/migrate/20200921173826_convert_repository_dependency_id_to_big_int.rb:8]: 🤣

**Review by jsonperl — APPROVED** [review]
I think this works. Prolly get a 👍 from @katzj too.

**Comment by katzj** [comment]
This needs to be done as a few PRs -- the first to disable reading which we merge and rollout. Then the actual migration (so that code isn't running still trying to read while the migration happens). And then reverting of the first.

**Pull request merged**